activerecord/lib/active_record/scoping/default.rb

```diff
@@ -115,7 +115,10 @@ def build_default_scope(base_rel = nil) # :nodoc:
             base_rel ||= relation
             evaluate_default_scope do
               default_scopes.inject(base_rel) do |default_scope, scope|
-                default_scope.merge(base_rel.scoping { scope.call })
+                scoped = base_rel.scoping do
+                  scope.is_a?(Proc) ? instance_exec(&scope) : scope.call
```

Comment:
Yes, any kind of object can be used in default scope if the object is callable. I saw this in Rails tests, for example [in this case](https://github.com/rails/rails/blob/master/activerecord/test/cases/scoping/default_scoping_test.rb#L37) it uses OpenStruct.
But now I think that I have to handle the `call` method case also in correct context. Change is on the way.